openstack/tools/upload_octavia_amphora_image.sh

```diff
@@ -1,3 +1,24 @@
 #!/bin/bash -eu
 
-juju run octavia-diskimage-retrofit/0 retrofit-image --wait 10m
+JUJU_VERSION=$(juju version | grep -oE '^[0-9]+' | head -1)
+
+if [ -z "$JUJU_VERSION" ]; then
+    echo "Error: Unable to determine Juju version."
+    exit 1
+fi
+
+if [ "$JUJU_VERSION" -ge 3 ]; then
+    if juju run octavia-diskimage-retrofit/leader retrofit-image --wait=10m; then
```

Comment:
Since the only difference between juju 2 and 3 is the name of the command verb, you could set the verb based on the juju version,

```bash
    run=run
    if juju ${run} octavia-diskimage-retrofit/leader retrofit-image --wait=10m; then
```

And condensed the change.